### ISSUE TITLE
v8::Function::NewInstance is deprecated

### ISSUE THREAD

**Issue by Jake-Gillberg** [phusion/node-sha3#29]
Line 59 of [addon.cpp](https://github.com/phusion/node-sha3/blob/master/src/addon.cpp#L59) uses a deprecated function. Update to maybe version to fix.

```cpp
Local<Function> cons = Nan::New<Function>(constructor);
info.GetReturnValue().Set(cons->NewInstance(argc, argv));
```
Compiler warning:
```
../src/addon.cpp: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE SHA3Hash::New(Nan::NAN_METHOD_ARGS_TYPE)':
../src/addon.cpp:59:58: warning: 'v8::Local<v8::Object> v8::Function::NewInstance(int, v8::Local<v8::Value>*) const' is deprecated (declared at /home/node/.node-gyp/7.7.3/include/node/v8.h:3657): Use maybe version [-Wdeprecated-declarations]
    info.GetReturnValue().Set(cons->NewInstance(argc, argv));
```

**Comment by Manish166** [comment]
Getting this same issue while adding web3.js in a Dapp.
Stack and versions are "web3": "^1.0.0-beta.30"
node - 8.9.4
npm - 5.6.0

./src/addon.cpp:59:36: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
                        info.GetReturnValue().Set(cons->NewInstance(argc, argv));
                                                        ^
/Users/manishsingh/.node-gyp/8.9.4/include/node/v8.h:3831:3: note: 'NewInstance' has been explicitly marked deprecated here
  V8_DEPRECATED("Use maybe version",
  ^
/Users/manishsingh/.node-gyp/8.9.4/include/node/v8config.h:321:29: note: expanded from macro 'V8_DEPRECATED'
  declarator __attribute__((deprecated))

**Comment by stasbar** [comment]
Same here, building on Docker

FROM node:latest
WORKDIR /usr/src/app
COPY package*.json ./
RUN npm install
COPY src/ src/
EXPOSE 8546
CMD [ "npm", "start" ]

package.json

**Comment by canterberry** [comment]
Fixed in [#33](https://github.com/phusion/node-sha3/pull/33).
